### DISCUSSION TITLE
Shift away animation for a reusable Tooltip component

### DISCUSSION THREAD

**Discussion by davegomez** [floating-ui/floating-ui#2432] [Q&A] [answered]
I am currently building a reusable Tooltip component with Floating UI. However, I am struggling to understand how to use the `useTransitionStyles` function to achieve the [swift-away-subtle](https://atomiks.github.io/tippyjs/#extra-included-animations) animation from Tippy.js.

In addition, when using CSS with Tailwind (our project's framework), the tooltip's initial position is at `0,0` of the window, causing the first animation to move the tooltip from the window's corner to the anchor's position. Although I can animate the `open` position with transform translate, I cannot get the initial position to work.

Furthermore, using the styles provided by `useTransitionStyles` has proven to be even more challenging as the tooltip always ends up being rendered far away from the anchor, typically at the bottom of the page, no matter what I do. I have attempted to recreate the animation using [Delay Groups](https://codesandbox.io/s/clever-aryabhata-yq3xyc?file=/src/Tooltip.tsx), and [Scale transforms](https://codesandbox.io/s/floating-ui-react-scale-transform-origin-qv0t1c?file=/src/App.tsx) examples, but with the same result.

I hope to find help understanding how to achieve the desired animation with Floating UI, and I appreciate any assistance you can provide.

**Comment by atomiks** [comment, marked as answer]
The floating element is positioned using `transform` styles by default. To add a transform animation without conflicting with the positioning, you use a wrapper:

```js
<div ref={refs.setFloating} style={floatingStyles}
  <div style={styles}>
    ...
  </div>
</div>
```

**Comment by davegomez** [comment]
This works @atomiks and I'd like to share a little snippet for future reference in case someone needs some help:

```jsx
const TooltipContent = React.forwardRef(
    ({ css, style, variant = "light", ...props }, ref) => {
        const context = useTooltipContext();
        const transition = useTransitionStyles(context, {
            initial: {
                opacity: 0,
                transform: "translateX(0) translateY(5px)",
            },
        });

        return (
            transition.isMounted && (
                <FloatingPortal>
                    <div
                        ref={context.refs.setFloating}
                        style={{ ...context.floatingStyles }}
                    >
                        <div
                            style={{
                                ...transition.styles,
                                ...style,
                            }}
                            {...context.getFloatingProps(props)}
                            ref={ref}
                        />
                    </div>
                </FloatingPortal>
            )
        );
    }
);
```

Thank you!

**Reply by romankoho** [reply]
Thank you, that was super helpful. I just couldn't animate the FloatingPortal until I found your solution
